### PR TITLE
Marked some formula as pre2018, and added residency from 2018 act.

### PR DESCRIPTION
**Pull request by Br3nda**
* Tax and benefit system evolution
* Impacted periods: from 2018
* Impacted areas: `variables/acts/social_security`
* Details:
  - Marked some formula as pre2018
  - Added residency from 2018 act
  - Found the commencement date for Job seeker (prior to April 2013 there was a different section "Unemployment benefit")

- - - -
- Impact the OpenFisca-Aotearoa public API
- Adds residency as per the 2018 social security act
- uses new residency in the job seeker formula

### CODE REVIEW

**Comment by Br3nda** [comment]
Note from conversation in Slack -- this is likely to trigger this issue soon
https://github.com/govzeroaotearoa/openfisca-aotearoa/issues/15

**Comment by Br3nda** [comment]
Note: The variables are all still set to `MONTH`

File: openfisca_aotearoa/tests/demographics/relationships.yaml [hunk 1]
```diff
@@ -5,19 +5,19 @@
...
         age:
-          "2018-10-01": 30
+          "2017-10-01": 30
       Fred:
         age:
-          "2018-10-01": 30
+          "2017-10-01": 30
       Mary:
         age:
-          "2018-10-01": 30
```
Comment (Br3nda) [inline openfisca_aotearoa/tests/demographics/relationships.yaml:20]: These tests were all over the commencement date of the new act, so i moved them back

**Review by piacandrews — APPROVED** [review]

**Review by bonjourmauko — APPROVED** [review]
✨

File: openfisca_aotearoa/variables/acts/social_security/74/residency.py [hunk 2]
```diff
@@ -39,7 +40,38 @@ def formula(persons, period, parameters):
...
+    #      the Immigration Act 2009, has resided continuously in New Zealand
+    #      for a period of at least 2 years at any one time,
+    def formula_2018_11_26(persons, period, parameters):
+
+        has_eligible_residency_class = persons("is_citizen_or_resident", period) + \
+            persons("immigration__is_recognised_refugee", period) + \
+            persons("immigration__is_protected_person", period)
+
+        nz_eligible = persons("social_security__is_ordinarily_resident_in_new_zealand", period) * persons("social_security__has_resided_continuously_in_nz_for_a_period_of_at_least_2_years_at_any_one_time", period)
+        reciprocal_eligible = persons("social_security__is_ordinarily_resident_in_country_with_reciprocity_agreement", period) * (persons("years_resided_continuously_in_new_zealand", period) > 2)
```
Comment (bonjourmauko) [inline openfisca_aotearoa/variables/acts/social_security/74/residency.py]: The text says _before_. By the current definition, that would mean at least 24 months as of last month.

Comment (Br3nda) [inline openfisca_aotearoa/variables/acts/social_security/74/residency.py]:
```suggestion
        reciprocal_eligible = persons("social_security__is_ordinarily_resident_in_country_with_reciprocity_agreement", period) * (persons("years_resided_continuously_in_new_zealand", period) >= 2)
```

**Comment by piacandrews** [comment]
Group decided that where we either make or find an interpretation (eg, from MSD website) which isn't in the legislation (such as ordinarily resident continuously = 24 months prior to current date) would be added into the coded as a #'d comment with reference to a URL, or identified as a call the team made.

**Review by piacandrews — APPROVED** [review]

File: openfisca_aotearoa/variables/acts/social_security/74/residency.py [hunk 3]
```diff
@@ -14,7 +14,8 @@ class social_security__meets_residential_requirements_for_certain_benefits(Varia
     definition_period = MONTH
     reference = "http://www.legislation.govt.nz/act/public/1964/0136/latest/DLM363796.html"
```
Comment (Br3nda) [inline openfisca_aotearoa/variables/acts/social_security/74/residency.py:15]:
```suggestion
    reference = "http://www.legislation.govt.nz/act/public/1964/0136/latest/DLM363796.html"
    documentation = """
    https://www.legislation.govt.nz/act/public/2018/0032/latest/whole.html#DLM6783138
    16 Residential requirement
    (1) This section sets out the residential requirement that must be met by applicants for certain types of assistance under this Part.
    (2) A person (P) meets the residential requirement if—
        (a) P is a New Zealand citizen or holds a residence class visa under the Immigration Act 2009, and is ordinarily resident in New Zealand when P
     first applies for the benefit, and—
            (i) has resided continuously in New Zealand for a period of at least 2 years at any one time after becoming a citizen or resident; or
            (ii) is recognised as a refugee or a protected person in New Zealand under the Immigration Act 2009; or
       (b) P is ordinarily resident in a country with which New Zealand has a reciprocity agreement, and P has resided continuously in New Zealand for a period of at least 2 years before applying for the benefit or before a decision on P’s claim for the benefit is made.
    (3) For the purposes of subsection (2)(b), New Zealand has a reciprocity agreement with another country if there is in force under section 380 an order declaring that the provisions contained in an agreement
     (for example, a convention) signed by New Zealand and the Government of that country set out in a schedule of the order have force and effect so far as they relate to New Zealand.
     (c) except in the case of a person who is recognised as a refugee or
         a protected person in New Zealand under
         the Immigration Act 2009, has resided continuously in New Zealand
         for a period of at least 2 years at any one time,
    """
```

**Review by MarkCalvert — APPROVED** [review]

**Pull request merged**